lib/cloud_controller/deployment_updater/actions/scale_down_superseded.rb

```diff
@@ -0,0 +1,30 @@
+module VCAP::CloudController
+  module DeploymentUpdater
+    module Actions
+      class ScaleDownSuperseded
```

Comment:
This might have to be Superseded+Canceled, as apparently Superseded+Deployed, is a separate case

Reply:
Do you know why we support superseded+deployed ?https://github.com/cloudfoundry/cloud_controller_ng/blob/f3834d2504f4b47de6cc23366d9483d484f1a8f7/app/actions/deployment_create.rb#L201 Seems like Superseded+Canceled is a more accurate state.

It seems like in the Superseded+Deployed case we don't scale down the deployment until we finalize the deployment https://github.com/cloudfoundry/cloud_controller_ng/blob/main/lib/cloud_controller/deployment_updater/updater.rb#L172C11-L172C30 . Is that because if our old deployment was half way through, we don't want to scale those down as we'll reduce the overall app instance number before we've rolled out our new deployment? 🤔

Reply:
>  Is that because if our old deployment was half way through, we don't want to scale those down as we'll reduce the overall app instance number before we've rolled out our new deployment? 

That's right, so instead those are finalized and their AI's are kept running.

I haven't quite wrapped my head around why/when we want to supersede + cancel (though looking at the code it seems like we just do that when we deploy over any other state, including prepaused/paused)

Reply:
Well, if you superseded over and over, the DeploymentUpdater will start to take down instances from those superseded deployments one by one as it brings up the latest/current deployment.

However, there is definitely a bug in the implementation.  You can definitely get an excessive number of running instances by having multiple superseding deployments.  See this comment for an example: https://github.com/cloudfoundry/cloud_controller_ng/pull/4101#issuecomment-2489827283